### ISSUE TITLE
How to handle untrusted output with real-time collaboration

### ISSUE THREAD

**Issue by ellisonbg** [jupyterlab/jupyterlab#12152]
### Problem

The Jupyter notebook architecture (jupyter server, notebook, lab, etc.) has an idea of trusted output. Here is the current documentation that describes how it has worked previously:

https://jupyter-notebook.readthedocs.io/en/stable/security.html#security-in-notebook-documents

To summarize that documentation:

* Untrusted HTML is always sanitized
* Untrusted Javascript is never executed
* HTML and Javascript in Markdown cells are never trusted
* Outputs generated by the user are trusted
* Any other HTML or Javascript (in Markdown cells, output generated by others) is never trusted
* The central question of trust is “Did the current user do this?”

Before RTC, there was a single user accessing a kernel and filesystem, so there wasn't ambiguity about who the user was. With RTC, multiple users can share a kernel, and the kernel doesn't know which user runs a given cell, so there is new ambiguity about how this should work. 

As JupyterLab now has real-time collaboration, the following questions are coming up:

* In the above documented security model, do we interpret "user" to mean "human user" or "kernel"?
* Should untrusted output from another user in the same notebook and a shared kernel be shown to me? Should I have to take action to see an untrusted output, or should it just show up always?
* At what point does an entire notebook become trusted? When all cells have been run by any user?

We should also run this by other Jupyter subprojects for input (hub, server, security).

**Comment by arogozhnikov** [comment]
Hi Brian,
I have the same issue even with working on the same notebook from pycharm and pycharm.

Pycharm's trust is project-based. You can simply trust all the notebooks in the project or not (which makes sense to me), origin of output doesn't make sense.

When I try to open same notebook in jupyterlab, there is no way right now even to specify "trust this notebook" or "trust this output". The only way I found is 'cd find/that/folder && jupyter trust ...'

Even more troublesome, it strips out css style sheets, even scoped ones. Though, maybe I should open a separate issue about the last one? I don't think that sanitizing should strip out scoped stylesheets.

**Comment by jasongrout** [comment]
More thoughts:

1. Multiple humans could always connect to the same kernel from different clients, but the other tooling around notebooks made this awkward, and didn't have the notion of "user", though you did have the notion of "stuff I executed" vs "stuff others executed". So in principle we had these sorts of issues before, though the tooling made it so awkward that I think it was rarely a problem in practice. RTC certainly makes these issues important to solve in practice.
2. @arogozhnikov - we had a menu item to trust a notebook, but it was removed a few years back (see https://github.com/jupyterlab/jupyterlab/issues/5354 and https://github.com/jupyterlab/jupyterlab/pull/5631). It is still available via the command palette (as the "Trust Notebook" command) or I think also clicking on the trust status bar icon.

**Comment by arogozhnikov** [comment]
> or I think also clicking on the trust status bar icon

Nope, does not. I think this is a result of some outputs being trusted and some not, but did not check.

Palette command works

**Comment by trungleduc** [comment]
@arogozhnikov `always trust notebook` and status bar icon action can be added by https://github.com/jupyterlab/jupyterlab/pull/11494
